File: app/src/main/java/com/lxk/animandcustomviewdemo/recyclerview/RecyclerViewActivity.java
package com.lxk.animandcustomviewdemo.recyclerview;

import android.content.pm.ActivityInfo;
import android.os.Bundle;
import android.view.View;

import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.lxk.animandcustomviewdemo.BaseClickActivity;
import com.lxk.animandcustomviewdemo.R;

import java.util.ArrayList;
import java.util.List;

/**
 * @author https://github.com/103style
 * @date 2019/11/14 16:29
 */
public class RecyclerViewActivity extends BaseClickActivity {

    private RecyclerView rv;
    private TestRecyclerView cRv;
    private TestItemDecoration itemDecoration = new TestItemDecoration();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_recyclerview);
        rv = findViewById(R.id.rv_test);
        cRv = findViewById(R.id.crv_test);

        setClickListener(
                R.id.linear_layout_manager,
                R.id.item_decoration,
                R.id.recycler_offset_children,
                R.id.recycler_layoutmanager,
                R.id.rv_3d
        );
    }

    private RecyclerView.Adapter getAdapter() {
        List<String> datas = new ArrayList<>();
        for (int i = 0; i < 50; i++) {
            datas.add(String.valueOf(i + 1));
        }
        return new TestAdapter(this, datas);
    }

    @Override
    public void onClick(View v) {
        setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);
        RecyclerView.LayoutManager layoutManager;
        rv.setVisibility(View.GONE);
        cRv.setVisibility(View.GONE);
        switch (v.getId()) {
            case R.id.item_decoration:
                rv.removeItemDecoration(itemDecoration);
                rv.addItemDecoration(itemDecoration);
                layoutManager = new LinearLayoutManager(this);
                rv.setLayoutManager(layoutManager);
                rv.setVisibility(View.VISIBLE);
                break;

            case R.id.linear_layout_manager:
                rv.removeItemDecoration(itemDecoration);
                layoutManager = new LinearLayoutManager(this);
                rv.setLayoutManager(layoutManager);
                rv.setVisibility(View.VISIBLE);
                break;

            case R.id.recycler_offset_children:
                layoutManager = new TestLayoutManager();
                rv.setLayoutManager(layoutManager);
                rv.setVisibility(View.VISIBLE);
                break;

            case R.id.recycler_layoutmanager:
                layoutManager = new TestLayout3Manager();
                rv.setLayoutManager(layoutManager);
                rv.setVisibility(View.VISIBLE);
                break;

            case R.id.rv_3d:
                setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE);
                layoutManager = new TestLayout4Manager();
                cRv.setLayoutManager(layoutManager);
                cRv.setVisibility(View.VISIBLE);
                cRv.setAdapter(getAdapter());
                break;

            default:
                break;
        }
        rv.setAdapter(getAdapter());

    }
}

File: app/src/main/java/com/lxk/animandcustomviewdemo/recyclerview/RecyclerViewActivity.kt
package com.lxk.animandcustomviewdemo.recyclerview

import android.content.pm.ActivityInfo
import android.os.Bundle
import android.view.View
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.lxk.animandcustomviewdemo.BaseClickActivity
import com.lxk.animandcustomviewdemo.R
import java.util.*

/**
 * @author https://github.com/103style
 * @date 2020/1/10 11:01
 */
class RecyclerViewActivity : BaseClickActivity() {

    private var rv: RecyclerView? = null
    private var cRv: TestRecyclerView? = null
    private val itemDecoration = TestItemDecoration()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_recyclerview)
        rv = findViewById(R.id.rv_test)
        cRv = findViewById(R.id.crv_test)

        setClickListener(
                R.id.linear_layout_manager,
                R.id.item_decoration,
                R.id.recycler_offset_children,
                R.id.recycler_layoutmanager,
                R.id.rv_3d
        )
    }

    private fun getAdapter(): TestAdapter {
        val data = ArrayList<String>()
        for (i in 1 until 50) {
            data.add(i.toString())
        }
        return TestAdapter(this, data)
    }

    override fun onClick(v: View) {
        requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
        val layoutManager: RecyclerView.LayoutManager?
        rv!!.visibility = View.GONE
        cRv!!.visibility = View.GONE
        when (v.id) {
            R.id.item_decoration -> {
                rv!!.removeItemDecoration(itemDecoration)
                rv!!.addItemDecoration(itemDecoration)
                layoutManager = LinearLayoutManager(this)
                rv!!.layoutManager = layoutManager
                rv!!.visibility = View.VISIBLE
            }
            R.id.linear_layout_manager -> {
                rv!!.removeItemDecoration(itemDecoration)
                layoutManager = LinearLayoutManager(this)
                rv!!.layoutManager = layoutManager
                rv!!.visibility = View.VISIBLE
            }
            R.id.recycler_offset_children -> {
                layoutManager = TestLayoutManager()
                rv!!.layoutManager = layoutManager
                rv!!.visibility = View.VISIBLE
            }
            R.id.recycler_layoutmanager -> {
                layoutManager = TestLayout3Manager()
                rv!!.layoutManager = layoutManager
                rv!!.visibility = View.VISIBLE
            }
            R.id.rv_3d -> {
                requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE
                layoutManager = TestLayout4Manager()
                cRv!!.layoutManager = layoutManager
                cRv!!.visibility = View.VISIBLE
                cRv!!.adapter = getAdapter()
            }
            else -> {
            }
        }
        rv!!.adapter = getAdapter()

    }
}
